### PR TITLE
Funding guide

### PR DESCRIPTION
**Pull request by Bucko13**

### CODE REVIEW

**Comment by nodar-chkuaselidze** [comment]
For demo this looks great !!!

But I wanted to express some ideas around fundraiser.

First when you subtract fixed amount of fee from the output in the beginning, it won’t work on production. Some inputs, multisig for example, will have much higher sizes than P2PK or P2PKH. Most likely they still will need split the coins, they might not want to switch to p2pkh inputs even in that case (especially if there’s a lot money going in there).

So instead of having `fundingTarget - fee` as output value, you would have `fundingTarget`. And all inputs would be required to submit their `targetAmount` + `fee` based on their input size. (this will be inconvenient because of the double fee in case of split, but more fair)… Also as a bonus, it won’t have limited number of inputs (at least predetermined), it will only be limited by TX Size limit and sigops.

File: guides-markdown/crowdfund-tx.md [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+- Add a fund matching scheme where someone can say they will match future contributions
+- Currently the examples split transactions to make a coin available that equals the target contribution amount. This is expensive since you have to broadcast multiple transactions. An interface to choose to donate from available available coins might help to make this more efficient.
+- This approach also isn't ideal if funders have non-p2pkh outputs. Something like a multisig will be much bigger and will cause the estimated fee to be too low. One possible approach would be to shift the burden of fees on the funders, have the application calculate the size of their contribution and add the resulting fee to their funding amount. Finally, use the amount calculated (including the fee) when [splitting the coins](#step-4-prepare-your-coins). This has the added advantage of not needing to limit the number of inputs
```
Comment (Bucko13) [inline guides-markdown/crowdfund-tx.md]: @nodar-chkuaselidze  Added this final point here.

**Comment by Bucko13** [comment]
@nodar-chkuaselidze Thinking more about your approach, I think it makes a lot of sense. I'm going to try and refactor to try this way. If it's too tedious then we can leave it as an option for another way for people to build it themselves :).

**Comment by Bucko13** [comment]
@nodar-chkuaselidze Made changes using your suggestions. Also updated the repo where the full code is hosted in case you wanted to check without all the documentation (it's linked in markdown)

**Pull request merged**